bindings/python/pymongoarrow/api.py

```diff
@@ -306,9 +386,11 @@ def _tabular_generator(tabular):
         for i in tabular.to_batches():
             for row in i.to_pylist():
                 yield row
-    elif DataFrame is not None and isinstance(tabular, DataFrame):
+    elif isinstance(tabular, pd.DataFrame):
         for row in tabular.to_dict("records"):
             yield row
+    elif isinstance(tabular, pl.DataFrame):
```

Comment:
We'll need a similar guard here to what we had for pandas

Reply:
When `pandas` was optional, there were no exceptions raised if one called `find_pandas_all`.
With `polars` I think we should. Instead of `pa.Table.to_pandas` which exists and will throw a decent exception, I am using `polars.DataFrame.from_arrow` which won't be available. What type of exception should I raise?

Reply:
Something like this one?

```
        raise ValueError(
            f"Invalid tabular data object of type {type(tabular)} \n"
            "Please ensure that it is one of the supported types: "
            "DataFrame, Table, or a dictionary containing NumPy arrays."
        )
```

Reply:
This doesn't feel elegant to me. Suggestions?
```
    if pl is None:
        raise ValueError("polars is not imported. Try pip install polars.")
```

Reply:
I've made changes and pushed.